src/common/cardstack/types.ts

```diff
@@ -215,3 +240,18 @@ export interface AlCardstackPage<EntityType=any>
     pages_remaining:number;
 }
 
+/**
+ *  This interfaces describes an active filter
+ */
+export interface AlCardstackActiveFilter<EntityType = any,PropertyType extends AlCardstackItemProperties=any>
+{
+    property:AlCardstackPropertyDescriptor;                 //  The property being filtered on
+    propField:string;                                       //  The DTO field the property's value can be found on
+    values:AlCardstackValueDescriptor[];                    //  The values being filtered for (descriptors)
+    rawValues:any[];                                        //  The values being filtered for (raw values)
+
+    /**
+     *  The default callback will return true ("match") if any of the values in the filter match a given entity's value for that field, but it can be overridden to provide custom logic.
+     */
+    callback:{(entity:EntityType,properties:PropertyType,filter:AlCardstackActiveFilter<EntityType,PropertyType>):boolean};
```

Comment:
It would be good to test a custom callback on a filter, too, come to think of it.

Reply:
We found today the need of customCallback for filter in portero app as some of the values are of type array.